### ISSUE TITLE
Search results count is too large for food-seekers scrolling single page — how to improve? Add pagination to search results?

### ISSUE THREAD

**Issue by fancyham** [hackforla/food-oasis#2216]
### Overview
**What did you observe or what is a likely problem?**

Right now, our search results say "604 results" which, while accurate in that that's the total number of matching results... while that's a nice bragging point for us, that's not a very useful number for food-seekers in their immediate task because it's not relevant to their current search (because they're unlikely to travel across LA to access a pantry on the other side of the county) for **nearby** food.

![image](https://github.com/user-attachments/assets/fae7b58b-276e-42ec-9b05-002c4404a6c6)

**What is the user trying to do?**

The user is probably trying to find nearby pantries.  So if we show, say, the nearest 20-30 to the searched address, that should be more than enough.  While it's possible a food-seeker might look at the 31st, but they'd probably want to explicitly expand their search area, or better, they'd probably simply move the location they're searching on.

**How many people are / will be having this problem?**

Currently, everyone on desktop.

### Discussion and possible Solution(s)
**What device is this happening on / what is the user’s context?**

Currently this only occurs on the Desktop -- the tablet portrait and mobile modes don't show total # of search results.

Since desktop has more space, we could perhaps show better information.

Something like "30 nearest locations (out of 604 total results)" when searching on a point address

Or, if searching for a region:

"30 nearest locations (there are [X] locations within "[RegionName]")

If we can add pages, that would also help users.  For example: Page 1 shows locations 1-10, page 2 shows 21-30, etc.

### Discussion and possible Solution(s)
**Possible challenges, solutions?**

- [ ] What's the range of possible solutions?

### Resources/Instructions
**How do others solve this?**
- [ ] Look at other web-based map websites like Yelp, Google, etc.

**Things to think about?**
What pieces of information are most useful for the user on this screen?

### Action items

- [ ] Discuss with Research and other designers, PMs, etc. what information is needed up there?
- [ ] Mock up alternatives for the results.
- [ ] Should there be pages of results?  On what screen sizes? (desktop only?) If pages, does that change the text? ("showing 1-10 of 604 results, sorted by distance").  If not pages, then what?
- [ ] Create designs and specifications with logic for devs
- [ ] Devs create beta
- [ ] Test it and revise
- [ ] Once it's good, launch

Also, consider a staged approach -- changing it little by little on the public site might be fine as long as devs are willing to make quick changes like that.

**Comment by fancyham** [comment]
Adding [John's Oct 14, 2024 mockups](https://www.figma.com/design/9hPmNSsyaH2VMGNz5mffmI/FOLA-Design-%232?node-id=8396-10760&t=XtOXb5YAU5ApxAG0-4):
![image](https://github.com/user-attachments/assets/7d708274-d0d2-4fc1-8f38-afeaf3f34cb9)


Hope it's okay -- I moved the design over to its own [Figma page #2116](https://www.figma.com/design/9hPmNSsyaH2VMGNz5mffmI/FOLA-Design-%232?node-id=640-132)

Thanks John, your designs spurred a good discussion!

Some thoughts we came up with during the meeting (this is a way more complex than it seems at first glance).

For one, the address that the user enters is currently auto-completed to a full address, so we'd be displaying: 
"Results for "Los Angeles, California 90015, United States" (23 locations)"
-- which can be a bit wordy.  

And we figured there are at least a few different flows that we need to account for, and each might have its own text variant:

1. User enters an address
2. User uses GPS button
3. User pans the map (currently there is no 'search here' button... might we want to add it?

Also, we were talking about what's actually useful for users... showing the 10 (or 20) closest locations to an address (or center of the map) seemed the most useful.  Then allow users to go to next closest set using pagination (or by panning map and clicking 'search here')

Which brings up these pagination and pin numbering/listings issues... it seems to be related... what users want are things near them... 10 seems like a good number to start with if we were to guess:
- [Feature: Visually indicate which item in the results list matches with which map pin]
- [Feature: Enable user to tell which item in the results list matches which map pin](https://github.com/hackforla/food-oasis/issues/1710) #1710

Shienny (added you to this issue so you get updates) said that she'll try to put together idea sketches for flows to show some possible options we discussed, also (she's going on vacation next week so will try her best))

**Comment by fancyham** [comment]
We don't really use size and complexity for design issues, but I'll add something anyway.

**Comment by ihorng** [comment]
I have been working on this and it's still work in progress. 

If we do introduce a pages of result, it should be available across all devices rather than limited to a specific screen size. Since this is a responsive experience, it makes sense for the messaging to stay consistent on both desktop and mobile. From a UX best-practice perspective, this follows a few key heuristics:

- Consistency and standards suggest users shouldn’t have to relearn how results work just because they’re on a different device. 
- Recognition rather than recall supports keeping the same model so users can easily understand what’s happening without extra effort. 
- Visibility of system status means the system should clearly communicate how and when results update, regardless of screen size. 

Notes from conversation from last year:
Meeting with Bryan: 1/23/25 
Decisions - Next steps:
Adding Pagination - 10 - 20 per page, start with 10

Here is the link to the figma file that I am working on: https://www.figma.com/design/9hPmNSsyaH2VMGNz5mffmI/FOLA-Design--2?node-id=640-132

**Comment by fancyham** [comment]
Another solution might be to show search results only for pins that are visible on the map.

This would reduce the number of search results to a more manageable and also intuitive number — probably around 30 at most for a neighborhood's worth of pins.

**Questions will be:**
- When do we update the search results.  Perhaps when user taps a 'search this area' button that appears when user moves map area, similar to many native mobile maps.
- Do we do this for desktop and mobile?  Mobile's map is so tiny, it may actually be worse than what we do now (show all 600+ results).  What happens when user resizes map/text split view?
